plugins/performance-lab/includes/admin/plugins.php

```diff
@@ -19,58 +19,66 @@
  * @return array{name: string, slug: string, short_description: string, requires: string|false, requires_php: string|false, requires_plugins: string[], download_link: string, version: string}|WP_Error Array of plugin data or WP_Error if failed.
  */
 function perflab_query_plugin_info( string $plugin_slug ) {
-	$plugin = get_transient( 'perflab_plugin_info_' . $plugin_slug );
-
-	if ( is_array( $plugin ) ) {
-		/**
-		 * Validated (mostly) plugin data.
-		 *
-		 * @var array{name: string, slug: string, short_description: string, requires: string|false, requires_php: string|false, requires_plugins: string[], download_link: string, version: string} $plugin
-		 */
-		return $plugin;
+	$transient_key = 'perflab_plugins_info';
+	$plugins       = get_transient( $transient_key );
+
+	if ( is_array( $plugins ) && isset( $plugins[ $plugin_slug ] ) ) {
+		return $plugins[ $plugin_slug ];
 	}
 
-	$fields = array(
-		'name',
-		'slug',
-		'short_description',
-		'requires',
-		'requires_php',
-		'requires_plugins',
-		'download_link',
-		'version', // Needed by install_plugin_install_status().
-	);
+	$request = wp_remote_get( 'https://api.wordpress.org/plugins/info/1.2/?action=query_plugins&request[author]=wordpressdotorg&request[tag]=performance&request[per_page]=100' );
```

Comment:
Why not use `plugins_api()` here?

Reply:
@westonruter Thanks for reviewing this PR. I have not used `plugins_api()` because this does not support passing multiple specific slugs (which would be ideal) .  Please let me know if I am missing anything.

Reply:
`plugins_api()` is a wrapper for any API calls to `https://api.wordpress.org/plugins/`, so you can use it here too. It's only the API itself that doesn't support passing multiple specific slugs, but we're not doing that here anyway.

You can call the `plugins_api()` function with the `query_plugins` action and provide the relevant request arguments.

Reply:
This seems to work fine:

```php
$response = plugins_api( 'query_plugins',
	array(
		'author'   => 'wordpressdotorg',
		'tag'      => 'performance',
		'per_page' => 100,
	)
);
```

I can see that `$response->plugins` is populated as expected.

Reply:
@felixarntz @westonruter Thanks for the feedback, I have implemnted the same.